chore(license): reuse compliance

fhilgers opened the pull request:
This commit adds SPDX Licenses and Copyright for every file in the repository. I tried to find out as best as I could to find the right Copyright holder for every previously existing file. For autogenerated files there is no copyright holder and the license is CC0-1.0. For non code files CC-BY-4.0 is used as specified by AMOS. For code files MIT is used as specified by AMOS.
For external code (gradle wrapper as example) their original license is used. As Catchall the original LICENSE file from AMOS is kept.

The reuse cli is added to the nix shell, `reuse lint` checks whether the project is compliant.
Signed-off-by: Felix Hilgers <felix.hilgers@fau.de>

Mr-Kanister reviewed (commented):
LGTM. Should we provide a small tutorial in the README on how to annotate new files with the fitting licenses or will this be handled automatically by CI?

fhilgers commented:
Good point, I will add that to the readme tomorrow. CI will only check whether it's compliant.

Mr-Kanister commented:
> Good point, I will add that to the readme tomorrow.

Perfect, I will approve this afterwards.

der-whity reviewed (dismissed):
LGTM

fhilgers commented:
I still have to add the readme entry

luca-dot-sh commented:
Can we also add this text somewhere? Brand Guidelines require it due to our logo.

"The Android robot is reproduced or modified from work created and shared by Google and used according to terms described in the [Creative Commons](https://creativecommons.org/licenses/by/3.0/) 3.0 Attribution License."

fhilgers commented:
I will look into the reuse documentation where this could be added. Sorry that this is taking so long. I am not at home until Sunday evening.

fhilgers commented:
I have

- Added a guide to the README (how to correctly set license headers)
- Added Logo to the README
- Added Footnote and Trademark for Android and logo
- Added correct license and attribution for our logo (see `Deliverables/sprint-01/team-logo.{svg,png}.license`). You can check that with `reuse spdx`, this will generate a report. For the logo files it has both `Google` and `The AMOS Projects` as the copyright holders with the `CC-BY-3.0` (Creative Commons Attribution 3.0).
- Rebased on Dev

Mr-Kanister reviewed (approved):
Perfect.
